### ISSUE TITLE
ruby: Cannot start multiple PhantomJS drivers concurrently

### ISSUE THREAD

**Issue by p0deje** [SeleniumHQ/selenium#1154]
An attempt to start multiple PhantomJS drivers concurrently maybe result in PhantomJS processes binding to the same port. Related to #736 

```ruby
require 'selenium-webdriver'

threads, drivers = [], []
2.times do
  threads << Thread.new do
    drivers << Selenium::WebDriver.for(:phantomjs)
  end
end

threads.each do |thread|
  thread.abort_on_exception = true
  thread.join
end

drivers.each_with_index do |driver, index|
  puts "Checking ##{index + 1}: #{driver}"
  puts "  it runs on #{driver.send(:bridge).instance_variable_get(:@service).instance_variable_get(:@uri).port}"

  driver.title # make any wire call
  driver.quit
end
```

```
bundle exec ruby test.rb
Checking #1: #<Selenium::WebDriver::Driver:0x007fbcf613c918>
  it runs on 8910
Checking #2: #<Selenium::WebDriver::Driver:0x007fbcf614acc0>
  it runs on 8910
/Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:879:in `initialize': Connection refused - connect(2) for "127.0.0.1" port 8910 (Errno::ECONNREFUSED)
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:879:in `open'
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:879:in `block in connect'
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/timeout.rb:74:in `timeout'
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:878:in `connect'
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:863:in `do_start'
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:852:in `start'
	from /Users/p0deje/.rbenv/versions/2.2.2/lib/ruby/2.2.0/net/http.rb:1375:in `request'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/default.rb:107:in `response_for'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/default.rb:58:in `request'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/http/common.rb:59:in `call'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/bridge.rb:664:in `raw_execute'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/bridge.rb:642:in `execute'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/remote/bridge.rb:197:in `getTitle'
	from /Users/p0deje/Development/selenium/rb/lib/selenium/webdriver/common/driver.rb:153:in `title'
	from test.rb:19:in `block in <main>'
	from test.rb:15:in `each'
	from test.rb:15:in `each_with_index'
	from test.rb:15:in `<main>'
```

**Comment by p0deje** [comment]
Fixed in f35a9dc7936009d30d32b86f3be7414e4b3beabc

**Comment by titusfortner** [comment]
@p0deje - fyi - the shared_example specs fail for all browsers on all platforms when $DEBUG = true with [process.io calls](https://github.com/SeleniumHQ/selenium/blob/master/rb/lib/selenium/webdriver/phantomjs/service.rb#L104)

**Comment by p0deje** [comment]
@titusfortner It was failing for PhantomJS and I fixed it with f4c1082af3e3e4db25fd207148d3f1d9ce24129a, but I cannot reproduce for other browsers.

**Comment by titusfortner** [comment]
Weird, I would swear I was getting errors elsewhere last night, but I'm not seeing them this morning. Thanks for figuring out the issue on phantomjs, I'm glad it was an easy fix (one I should have been able to figure out myself. IO stuff is weird. I can't get Edge tests working without allowing IO output. Peculiar. Plus Jari's Child Process gem is mostly over my head. :)
¯¯\\__(ツ)__/¯¯
